src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -105,7 +79,7 @@ def get_packages_to_cover(
         Addresses((adaptor.address,))
     )
     all_targets = transitive_hydrated_targets.closure
-
+    plugin_file_digest = await Get[Digest](InputFilesContent, get_coverage_plugin_input())
```

Comment:
I recommend only doing this when coverage is used as a slight performance improvement.

Generally, you're using `test_options.values.run_coverage` in multiple places so will want to extract it into a variable like `use_coverage = test_options.values.run_coverage`.